src/circuit/gadget/ecc.rs

```diff
@@ -376,3 +378,332 @@ impl<C: CurveAffine, EccChip: EccInstructions<C> + Clone + Debug + Eq> FixedPoin
+                        if C::from_xy(x, y).is_some().into() {
+                            assert_eq!(real_added.to_affine(), C::from_xy(x, y).unwrap());
+                        }
+                    }
+                }
+            }
+
+            // Check incomplete addition
+            {
+                // P + Q
+                let real_added = p_val + q_val;
+                let added_incomplete = p.add_incomplete(layouter.namespace(|| "P + Q"), &q)?;
+                if let (Some(x), Some(y)) = (
+                    added_incomplete.inner.x.value,
+                    added_incomplete.inner.y.value,
+                ) {
+                    if C::from_xy(x, y).is_some().into() {
+                        assert_eq!(real_added.to_affine(), C::from_xy(x, y).unwrap());
+                    }
+                }
+
+                // P + P should return an error
+                p.add_incomplete(layouter.namespace(|| "P + P"), &p)
+                    .expect_err("P + P should return an error");
+
+                // P + (-P) should return an error
+                p.add_incomplete(layouter.namespace(|| "P + (-P)"), &p_neg)
+                    .expect_err("P + (-P) should return an error");
+
+                // P + 𝒪 should return an error
+                p.add_incomplete(layouter.namespace(|| "P + 𝒪"), &zero)
+                    .expect_err("P + 0 should return an error");
+
+                // 𝒪 + P should return an error
+                zero.add_incomplete(layouter.namespace(|| "𝒪 + P"), &p)
+                    .expect_err("0 + P should return an error");
+
+                // 𝒪 + 𝒪 should return an error
+                zero.add_incomplete(layouter.namespace(|| "𝒪 + 𝒪"), &zero)
+                    .expect_err("𝒪 + 𝒪 should return an error");
+            }
+
+            // Check fixed-base scalar multiplication
+            {
+                let scalar_fixed = C::Scalar::rand();
+                let nullifier_k = constants::nullifier_k::generator();
+                let base = nullifier_k.0.value();
+                let real_mul_fixed = base * scalar_fixed;
+
+                let scalar_fixed = super::ScalarFixed::new(
+                    chip.clone(),
+                    layouter.namespace(|| "ScalarFixed"),
+                    Some(scalar_fixed),
+                )?;
+                let nullifier_k = super::FixedPoint::get(
+                    chip.clone(),
+                    OrchardFixedBases::NullifierK(nullifier_k),
+                )?;
+
+                // [a]B
+                let mul_fixed = nullifier_k.mul(layouter.namespace(|| "mul"), &scalar_fixed)?;
+                if let (Some(x), Some(y)) = (mul_fixed.inner.x.value, mul_fixed.inner.y.value) {
+                    assert_eq!(real_mul_fixed.to_affine(), C::from_xy(x, y).unwrap());
+                }
+
+                // [0]B should return an error since fixed-base scalar multiplication
+                // uses incomplete addition internally.
```

Comment:
This may be a problem, because we require at least the signed 64-bit fixed-base scalar multiplication to work correctly for v = 0.

Reply:
Fixed-base scalar multiplication now uses complete addition on the last step, so it returns `(0,0)` for `[0]B`.